README

```diff
@@ -668,42 +668,93 @@ Network Support
+
+  1. If InfiniBand devices are available, use the UCX PML.
+
+  2. If PSM, PSM2, or other tag-matching-supporting Libfabric
+     transport devices are available (e.g., Cray uGNI), use the "cm"
+     PML and a single appropriate corresponding "mtl" module.
+
+  3. If MXM/InfiniBand devices are availble, use the "yalla" PML
+     (NOTE: the "yalla"/MXM PML is deprecated -- see below).
+
+  4. Otherwise, use the ob1 PML and one or more appropriate "btl"
+     modules.
+
+  Users can override Open MPI's default selection algorithms and force
+  the use of a specific transport if desired by setting the "pml" MCA
+  parameter (and potentially the "btl" and/or "mtl" MCA parameters) at
+  run-time:
+
+      shell$ mpirun --mca pml ob1 --mca btl [comma-delimted-BTLs] ...
+      or
+      shell$ mpirun --mca pml cm --mca mtl [MTL] ...
       or
-      shell$ mpirun --mca pml cm ...
-
-- Similarly, there are two OpenSHMEM network models available: "ucx",
-  and "ikrit":
-  - "ucx" interfaces directly with UCX;
-  - "ikrit" interfaces directly with Mellanox MXM.
-
-- UCX is the Unified Communication X (UCX) communication library
-  (http://www.openucx.org/).
-  This is an open-source project developed in collaboration between
-  industry, laboratories, and academia to create an open-source
-  production grade communication framework for data centric and
-  high-performance applications.
-  UCX currently supports:
-  - OFA Verbs;
-  - Cray's uGNI;
-  - NVIDIA CUDA drivers.
-
-- MXM is the Mellanox Messaging Accelerator library utilizing a full
-  range of IB transports to provide the following messaging services
-  to the upper level MPI/OpenSHMEM libraries:
-
-  - Usage of all available IB transports
-  - Native RDMA support
-  - Progress thread
-  - Shared memory communication
-  - Hardware-assisted reliability
+      shell$ mpirun --mca pml ucx ...
+
+  As alluded to above, there is actually a fourth MPI point-to-point
+  transport, but it is deprecated and will likely be removed in a
+  future Open MPI release:
+
+  - "yalla" uses the Mellanox MXM transport library.  MXM is the
+     deprecated Mellanox Messaging Accelerator library, utilizing a
+     full range of IB transports to provide the following messaging
+     services to the upper level MPI/OpenSHMEM libraries.  MXM is only
+     included in this release of Open MPI for backwards compatibility;
+     the "ucx" PML should be used insead.
+
+- The main OpenSHMEM network model is "ucx"; it interfaces directly
+  with UCX.
+
+  The "ikrit" OpenSHMEM network model is also available, but is
+  deprecated; it uses the deprecated Mellanox Message Accelerator
+  (MXM) library.
+
+- In prior versions of Open MPI, InfiniBand and RoCE support was
+  provided through the openib BTL and ob1 PML plugins.  Starting with
+  Open MPI 4.0.0, InfiniBand support through the openib+ob1 plugins is
+  both deprecated and superseded by the ucx PML component.
+
```

Comment:
Not entirely true. ob1 use is not deprecated, just btl/openib. There are cases where pml/ob1 + uct,vader,self will beat pml/ucx. Same is true for osc/rdma + btl/uct vs osc/ucx.

Reply:
Good catch. However this is for 4.0 and I was under the impression that uct BTL + OB1 is not yet ready for primetime (outside master) ?

Reply:
After some discussion on the webex, I'll remove the `+ob1` part of the clause.

Reply:
Fixed in #6027; will cherry pick all the #6027 commits over here to this v4.0.x branch PR when it is merged to master.